### PR TITLE
PSTRESS-56 Add support for ALGORITHM=INPLACE|COPY|INSTANT with DROP,TRUNCATE PARTITION

### PR DESCRIPTION
**Pull request by mohitj1988**

### CODE REVIEW

File: src/random_test.cpp [hunk 1]
```diff
@@ -252,7 +252,7 @@ inline static std::string pick_algorithm_lock() {
   static auto lock = opt_string(LOCK);
   static auto algorithm = opt_string(ALGORITHM);
   std::string locks[] = {"DEFAULT", "EXCLUSIVE", "SHARED", "NONE"};
-  std::string algorithms[] = {"INPLACE", "COPY", "DEFAULT"};
```
Comment (rahulmalik87) [inline src/random_test.cpp:255]: why you are removing default algorithms, It is valid syntax for most of ALTER. if you want you can add INSTANT.

File: src/random_test.cpp [hunk 2]
```diff
@@ -976,7 +976,8 @@ void Table::Analyze(Thd1 *thd) {
 
 void Table::Truncate(Thd1 *thd) {
   /* 99% truncate the some partition */
-  if (type == PARTITION && rand_int(100) > 98) {
```
Comment (rahulmalik87) [inline src/random_test.cpp:979]: existing problem. it should be 1 instead of 98. based on the above comment. please fix it as part of this patch.

**Review by rahulmalik87 — CHANGES_REQUESTED** [review]

File: src/random_test.cpp (hunk 1 above)
Comment (mohitj1988) [reply src/random_test.cpp:255]: The reason for removing "default" is that the ALGORITHM default value is INPLACE which is already covered.

File: src/random_test.cpp (hunk 2 above)
Comment (mohitj1988) [reply src/random_test.cpp:979]: Ok. Do we need to retain the OR condition I have added ?

File: src/random_test.cpp (hunk 1 above)
Comment (rahulmalik87) [reply src/random_test.cpp:255]: no. better to retain it.  you can add one more if you want. but please verify it works for all combinations.

**Review by rahulmalik87 — CHANGES_REQUESTED** [review]

**Review by rahulmalik87 — APPROVED** [review]
please ensure, there are no syntax error during 5 min run on will all ddls

**Pull request merged**